src/handle_locator/locate_handle.cpp

```diff
@@ -0,0 +1,228 @@
+#include <ros/ros.h>
+#include <actionlib/server/simple_action_server.h>
+#include <tue_msgs/LocateDoorHandleAction.h>
+#include <pcl_ros/transforms.h>
+#include <pcl_conversions/pcl_conversions.h>
+
+#include <pcl/filters/passthrough.h>
+#include <pcl/filters/voxel_grid.h>
+#include <pcl/filters/extract_indices.h>
+#include <pcl/search/kdtree.h>
+#include <pcl/segmentation/sac_segmentation.h>
+#include <pcl/segmentation/extract_clusters.h>
+#include <pcl/common/common.h>
+
+#include <pcl_ros/point_cloud.h>
+#include <pcl/point_types.h>
+
+typedef pcl::PointCloud<pcl::PointXYZ> PointCloud;
+
+actionlib::SimpleActionServer<tue_msgs::LocateDoorHandleAction>* as_;
+std::shared_ptr<tf::TransformListener> listener_;
```

Comment:
```suggestion
std::shared_ptr<tf::TransformListener> tf_listener_;
```

When applying, change the usage in the rest of code

Reply:
I suggest to use an `std::unique_ptr`